### PR TITLE
Add separator arguments to `Doc.getRange` and `Doc.getValue`

### PR DESCRIPTION
**Pull request by just95**
`Doc.getRange` and `Doc.getValue` accept an optional `separator` argument to specify the string to be used to separate lines.

### CODE REVIEW

File: lib/codemirror.dart [hunk 1]
```diff
@@ -625,8 +625,15 @@ class Doc extends ProxyHolder {
     return _editor;
   }
 
-  String getValue() => call('getValue');
+  /**
+   * Get the current editor content. You can pass it an optional argument to
+   * specify the string to be used to separate lines (defaults to "\n").
+   */
+  String getValue([String separator = '\n']) => callArg('getValue', separator);
```
Comment (devoncarew) [inline lib/codemirror.dart]: We should probably not set a default for `separator` then, and let codemirror set it, if `separator` is null.

File: lib/codemirror.dart [hunk 2]
```diff
@@ -876,8 +883,8 @@ class Doc extends ProxyHolder {
    * argument can be given to indicate the line separator string to use
    * (defaults to "\n").
    */
-  String getRange(Position from, Position to) {
-    return callArgs('getRange', [from.toProxy(), to.toProxy()]);
+  String getRange(Position from, Position to, [String separator = '\n']) {
```
Comment (devoncarew) [inline lib/codemirror.dart]: ditto, let's fall through to the codemirror default

**Comment by just95** [comment]
Good point. I have removed the default values for both arguments.

**Comment by devoncarew** [comment]
lgtm; thanks

**Pull request merged**